src/vsphere_cpi/spec/integration/attach_tag_to_vm_spec.rb

```diff
@@ -0,0 +1,313 @@
+require 'integration/spec_helper'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+require 'ostruct'
+
+module VSphereCloud
+  describe TaggingTag::AttachTagToVm, fake_logger: true do
+    before(:all) do
+      @host = fetch_property('BOSH_VSPHERE_CPI_HOST')
+      @user = fetch_property('BOSH_VSPHERE_CPI_USER')
+      @password = fetch_property('BOSH_VSPHERE_CPI_PASSWORD')
+      create_cat_and_tag
+    end
+
+    after(:all) do
+      delete_category
```

Comment:
Should this order be reversed?

Reply:
Also, maybe have a single function here `delete_cat_and_tag` as you did for creating

Reply:
@EleanorRigby You are right, I have updated the code implementation.